src/toil/test/src/threadingTest.py

```diff
@@ -0,0 +1,145 @@
+            log.info('Iteration %d', it)
+        
+            scope = self._createTempDir()
+            arena_name = 'thunderdome'
+            # Use processes (as opposed to threads) to prevent GIL from ordering things artificially
+            pool = multiprocessing.Pool()
+            try:
+                numTasks = 100
+                results = pool.map_async(
+                    func=partial(_testLastProcessStandingTask, scope, arena_name),
+                    iterable=list(range(numTasks)))
+                results = results.get()
+            finally:
+                pool.close()
+                pool.join()
+            
+            self.assertEqual(len(results), numTasks)
+            for item in results:
+                # Make sure all workers say they succeeded
+                self.assertEqual(item, True)
+            for filename in os.listdir(scope):
+                assert not filename.startswith('precious'), "File {} still exists".format(filename)
+
+def _testGlobalMutexOrderingTask(scope, mutex, number):
+    try:
+        # We will all fight over the potato
+        potato = os.path.join(scope, 'potato')
+        
+        with global_mutex(scope, mutex):
+            log.info('PID %d = num %d running', os.getpid(), number)
+            assert not os.path.exists(potato), "We see someone else holding the potato file"
+            
+            # Put our name there
+            out_stream = open(potato, 'w')
+            out_stream.write(str(number))
+            out_stream.close()
+            
+            # Wait
+            time.sleep(random.random() * 0.01)
+            
+            # Make sure our name is still there
+            in_stream = open(potato, 'r')
+            seen = in_stream.read().rstrip() 
+            assert seen == str(number), "We are {} but {} stole our potato!".format(number, seen)
+            in_stream.close()
+            
+            os.unlink(potato)
+            assert not os.path.exists(potato), "We left the potato behind"
+            log.info('PID %d = num %d dropped potato', os.getpid(), number)
+        return True
+    except:
+        traceback.print_exc()
+        return False
+        
+def _testLastProcessStandingTask(scope, arena_name, number):
+    try:
+        arena = LastProcessStandingArena(scope, arena_name)
+        
+        arena.enter()
+        log.info('PID %d = num %d entered arena', os.getpid(), number)
+        try:
+            # We all make files
+            my_precious = os.path.join(scope, 'precious' + str(number))
+            
+            # Put our name there
+            out_stream = open(my_precious, 'w')
+            out_stream.write(str(number))
+            out_stream.close()
+            
+            # Wait
+            time.sleep(random.random() * 0.01)
+            
+            # Make sure our file is still there unmodified
+            assert os.path.exists(my_precious), "Precious file {} has been stolen!".format(my_precious)
```

Comment:
Mostly a nit, and somewhat DRY, but this code seems to be repeating the potato experience above.  Would it be useful to be centralized in a single, say precious potato, function they will both be forced to share?

Reply:
Maybe if we use it again. I kept it separate because even though the contents logic is the same, when the files are created and destroyed is different, and a write-wait-read-back function seemed a bit of a weird unit.